src/Tools/Source/RunTests/Program.cs

```diff
@@ -377,5 +383,28 @@ private static IOrderedEnumerable<string> OrderAssemblyList(IEnumerable<string>
                 Logger.Log("Unable to send results");
             }
         }
+
+        /// <summary>
+        /// Checks the total size of dump file and removes files exceeding a limit.
+        /// </summary>
+        private static void CheckTotalDumpFilesSize()
+        {
+            var directory = Directory.GetCurrentDirectory();
+            var dumpFiles = Directory.EnumerateFiles(directory, "*.dmp", SearchOption.AllDirectories).ToArray();
+            long currentSize = 0;
+
+            foreach(var dumpFile in dumpFiles)
+            {
+                long fileSizeInMegabytes = (new FileInfo(dumpFile).Length / 1024) / 1024;
+                if (currentSize + fileSizeInMegabytes > MaxTotalDumpSizeInMegabytes)
+                {
+                    File.Delete(dumpFile);
+                }
+                else
+                {
+                    currentSize += fileSizeInMegabytes;
```

Comment:
I see what you mean. I thought that it confuses the meaning of the currentSize variable. Let us follow your idea and also rename the variable.

---
In reply to: [122765604](https://github.com/dotnet/roslyn/pull/20129#discussion_r122765604) [](ancestors = 122765604)